gazelle_haskell_modules/rule_generation.go

```diff
@@ -398,17 +398,29 @@ type ModuleImport struct {
 	ModuleName  string
 }
 
+type TempModuleImport struct {
+	IsSourceImported bool
+	ModuleName       []string
+}
+
 func (moduleImport *ModuleImport) UnmarshalJSON(data []byte) error {
-	var aux []string
-	if err := json.Unmarshal(data, &aux); err != nil {
+	var tempMod TempModuleImport
+	if err := json.Unmarshal(data, &tempMod); err != nil {
 		return err
 	}
+	aux := tempMod.ModuleName
 	pkgName := ""
 	if len(aux) > 1 {
 		pkgName = aux[0]
 	}
+	suffix := ""
+	// We add the suffix `.hs-boot` to the module name,
+	// when this module comes from an `hs-boot` file.
+	if tempMod.IsSourceImported {
+		suffix = ".hs-boot"
+	}
 	moduleImport.PackageName = pkgName
-	moduleImport.ModuleName = aux[len(aux)-1]
+	moduleImport.ModuleName = aux[len(aux)-1] + suffix
```

Comment:
Did you consider permanently storing the `IsSourceImported` field in `ModuleImport` instead of adding it to the string and using that as the future source of information? It might be easier to understand/not forget in the long run, albeit possibly more of a hassle to implement.